### ISSUE TITLE
Can't build for iOS - Cordova 6.0.0 & cordova-ios@4.0

### ISSUE THREAD

**Issue by alejandromagnorsky** [mauron85/cordova-plugin-background-geolocation#28]
Using Cordova iOS 4.0.1 I'm getting this error when running ```sudo cordova build ios```:
```
Plugins/cordova-plugin-mauron85-background-geolocation/CDVBackgroundGeoLocation.m:8:9: fatal error: 'Cordova/CDVJSON.h' file not found
#import <Cordova/CDVJSON.h>
```

**Comment by mauron85** [comment]
Try upgrading cordova to latest version. 4.0.1 is outdated.
```
 npm install -g cordova
```

**Comment by alejandromagnorsky** [comment]
I'm using cordova v5.4.1 and cordova iOS v4.0.1 (https://cordova.apache.org/announcements/2015/12/18/cordova-ios-4.0.1.html)

**Comment by mauron85** [comment]
@alejandromagnorsky I've just upgraded ios platform to v4.0.1 and yes it's failing. 

Downgrading to platform v3.9.2 should resolve your problem:
```
cordova platform update ios@3.9.2
```

One remark, why you're using sudo with cordova build?

**Comment by mauron85** [comment]
Output for future reference

```
The following build commands failed:
	CompileC build/Cordova\ Background\ GeoLocation.build/Debug-iphonesimulator/Cordova\ Background\ GeoLocation.build/Objects-normal/i386/MainViewController.o Cordova\ Background\ GeoLocation/Classes/MainViewController.m normal i386 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
ERROR building one of the platforms: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/cordova/build-debug.xcconfig,-project,Cordova Background GeoLocation.xcodeproj,ARCHS=i386,-target,Cordova Background GeoLocation,-configuration,Debug,-sdk,iphonesimulator,build,VALID_ARCHS=i386,CONFIGURATION_BUILD_DIR=/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/build/sharedpch
You may not have the required environment or OS to build this project
Error: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/cordova/build-debug.xcconfig,-project,Cordova Background GeoLocation.xcodeproj,ARCHS=i386,-target,Cordova Background GeoLocation,-configuration,Debug,-sdk,iphonesimulator,build,VALID_ARCHS=i386,CONFIGURATION_BUILD_DIR=/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/finch/dev/cordova-plugin-background-geolocation/example/SampleApp/platforms/ios/build/sharedpch
```

**Comment by mauron85** [comment]
This is just platform update which seems not working with this plugin.
We should wait for complete cordova update. If plugin stop working then, I'll look for ways how to fix it.

**Comment by alejandromagnorsky** [comment]
Ironically, cordova updated a few hours ago to version 6.0.0 and now it uses by default cordova iOS ~4.0.0 haha.

Note: About `sudo`, I reinstalled npm in my computer properly so I don't need `sudo` anymore ;)

**Comment by mauron85** [comment]
@alejandromagnorsky did you test the plugin? does it work?

**Comment by alejandromagnorsky** [comment]
@mauron85 Now I'm getting this error (due to the last update to this library):
```
/Plugins/cordova-plugin-geolocation-ios9-fix/CDVLocation.m:21:9: fatal error: 'Cordova/NSArray+Comparisons.h' file not found
#import <Cordova/NSArray+Comparisons.h>
```
If I use your plugin v0.9.1 it builds successfully.

**Comment by mauron85** [comment]
Can you try with clean sheet?

```
cd cordova
rm -rf plugins/ platforms/
cordova platform add ios
cordova build ios
```

**Comment by mauron85** [comment]
Welcome back :-(

**Comment by alejandromagnorsky** [comment]
OK, I will stick for now to the old versions.

**Comment by mauron85** [comment]
Apparently they didn't. Here is the diff of CDVLocation.m. And they still missing ```allowsBackgroundLocationUpdates = YES;```, which is I believe root of the problems. Why is the mystery. There is already pull [request for it](https://github.com/apache/cordova-plugin-geolocation/pull/51). Seems to be ignored.

```
21d20
< #import <Cordova/NSArray+Comparisons.h>
31,38d29
< //Edited by kingalione: START
< #define SYSTEM_VERSION_EQUAL_TO(v)                  ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedSame)
< #define SYSTEM_VERSION_GREATER_THAN(v)              ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedDescending)
< #define SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(v)  ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] != NSOrderedAscending)
< #define SYSTEM_VERSION_LESS_THAN(v)                 ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedAscending)
< #define SYSTEM_VERSION_LESS_THAN_OR_EQUAL_TO(v)     ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] != NSOrderedDescending)
< //Edited by kingalione: END
<
65,83c56,63
<     - (CDVPlugin*)initWithWebView:(UIWebView*)theWebView
<     {
<         self = (CDVLocation*)[super initWithWebView:(UIWebView*)theWebView];
<         if (self) {
<             self.locationManager = [[CLLocationManager alloc] init];
<
<             //Edited by kingalione: START
<             if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"9.0")) {
<                 self.locationManager.allowsBackgroundLocationUpdates = YES;
<             }
<             //Edited by kingalione: END
<
<             self.locationManager.delegate = self; // Tells the location manager to send updates to this object
<             __locationStarted = NO;
<             __highAccuracyEnabled = NO;
<             self.locationData = nil;
<         }
<         return self;
<     }
---
> - (void)pluginInitialize
> {
>     self.locationManager = [[CLLocationManager alloc] init];
>     self.locationManager.delegate = self; // Tells the location manager to send updates to this object
>     __locationStarted = NO;
>     __highAccuracyEnabled = NO;
>     self.locationData = nil;
> }
145,148c125,128
<         if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"]){
<             [self.locationManager requestAlwaysAuthorization];
<         } else if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationWhenInUseUsageDescription"]) {
<             [self.locationManager  requestWhenInUseAuthorization];
---
>         if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationWhenInUseUsageDescription"]){
>             [self.locationManager requestWhenInUseAuthorization];
>         } else if([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"]) {
>             [self.locationManager  requestAlwaysAuthorization];
171d150
<         // TODO: Set distance filter to 10 meters? and desired accuracy to nearest ten meters? arbitrary.
173c152
<         self.locationManager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters;
---
>         self.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers;
```

**Comment by mauron85** [comment]
Fixed in 0.9.3 with many other things. Check CHANGES.md. Reopen if necessary.
